src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -10933,6 +10933,15 @@ private static bool IsNullabilityMismatch(TypeSymbol type1, TypeSymbol type2)
         public override BoundNode? VisitQueryClause(BoundQueryClause node)
         {
             var result = base.VisitQueryClause(node);
+
+            if (node.Value is BoundLambda lambda)
+            {
+                // 'VisitConversion' only visits a lambda when the lambda has an AnonymousFunction conversion.
+                // This lambda doesn't have a conversion, so we need to visit it here.
+                Debug.Assert(node.HasAnyErrors);
```

Comment:
Why do we need the `Debug.Assert` call here? Is that because this is there is no conversion only when there are errors?

Reply:
> there is no conversion only when there are errors?

Yes. I got inspired here:

https://github.com/dotnet/roslyn/blob/ef5680468c491a5e7274ce68c944fb1c601b20eb/src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs#L6215-L6223